lib/leche.js

```diff
@@ -98,6 +98,42 @@ function createObject(proto) {
 }
 
 /**
+ * Returns the first maxLen characters of a the JSON string representation of
+ * the given object.
+ *
+ * @param {Object} object The object to get a string representation for.
+ * @param {number} maxLen The number of characters to truncate to.
+ * @returns {string} The first maxLen characters of the JSON string.
+ */
+function truncatedJSONStringify(object, maxLen) {
+	return JSON.stringify(object).slice(0, maxLen);
+}
+
+/**
+ * Gets a string representation of an object. If there is a usable toString()
+ * implementation, then it will return that. Otherwise, it will attempt to get
+ * a more specific string representation by getting the JSON string version of
+ * the object. For arrays, it will attempt to stringify its items recursively.
+ *
+ * @param {*} object The object to get a string representation for.
+ * @param {number} maxDepth The maximum depth to recurse for arrays.
+ * @returns {string} The string representation of the object.
+ * @private
+ */
+function stringifyObject(object, maxDepth) {
+	var stringRepresentation = object.toString();
+	if (object instanceof Array && maxDepth > 0) {
+		return object.map(function(item) {
+			return stringifyObject(item, maxDepth - 1);
+		}).toString();
+	} else if (stringRepresentation === '[object Object]') {
+		return truncatedJSONStringify(object, 30);
```

Comment:
We may still get the same issue if the first 30 characters of a JSON representation of 2 objects is the same. Leave a comment here? Not sure how rare that would be but should make sure it doesn't bite us in the future.
```
[
  { a: '1234567890': b: '1234567890', c: '1234567890', d: '1234567890' },
  { a: '1234567890': b: '1234567890', c: '1234567890', e: '1234567890' }
]
```